src/geocel/VolumeParams.cc

```diff
@@ -0,0 +1,83 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file geocel/VolumeParams.cc
+//---------------------------------------------------------------------------//
+#include "VolumeParams.hh"
+
+#include "inp/Model.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from input.
+ */
+VolumeParams::VolumeParams(inp::Volumes const& in)
+{
+    CELER_VALIDATE(in, << "no volumes were available");
+
+    // Build label maps
+    auto extract_labels = [](auto const& items) {
+        std::vector<Label> labels;
+        labels.reserve(items.size());
+        for (auto const& item : items)
+        {
+            labels.push_back(item.label);
+        }
+        return labels;
+    };
+    v_labels_ = VolumeMap("volume", extract_labels(in.volumes));
+    vi_labels_
+        = VolInstMap("volume_instance", extract_labels(in.volume_instances));
+
+    // Unzip volume properties
+    materials_.resize(in.volumes.size());
```

Comment:
```suggestion
    materials_.resize(this->num_volumes());
```
Just helps readability to be consistent